plugins/default/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -0,0 +1,797 @@
+                            , "lenses."
+                            ]
+
+                return $ List lenses
+  where
+    trivial (Range p p') = p == p'
+
+evalCommandName :: CommandId
+evalCommandName = "evalCommand"
+
+evalCommand :: PluginCommand
+evalCommand = PluginCommand evalCommandName "evaluate" runEvalCmd
+
+-- |Specify the test section to execute
+data EvalParams = EvalParams
+    { sections :: [Section]
+    , module_ :: !TextDocumentIdentifier
+    }
+    deriving (Eq, Show, Generic, FromJSON, ToJSON)
+
+runEvalCmd :: CommandFunction EvalParams
+runEvalCmd lsp st EvalParams{..} =
+    let dbg = logWith st
+        perf = timed dbg
+        cmd = do
+            let tests = testsBySection sections
+
+            let TextDocumentIdentifier{_uri} = module_
+            fp <- handleMaybe "uri" $ uriToFilePath' _uri
+            let nfp = toNormalizedFilePath' fp
+            mdlText <- moduleText lsp _uri
+
+            session <- runGetSession st nfp
+
+            (ms, _) <-
+                liftIO $
+                    runAction "runEvalCmd.getModSummary" st $
+                        use_ GetModSummary nfp
+
+            now <- liftIO getCurrentTime
+
+            let modName = moduleName $ ms_mod ms
+                thisModuleTarget =
+                    Target
+                        (TargetFile fp Nothing)
+                        False
+                        (Just (textToStringBuffer mdlText, now))
+
+            -- Setup environment for evaluation
+            hscEnv' <-
+                withSystemTempFile (takeFileName fp) $ \logFilename logHandle ->
+                    ExceptT $
+                        (either Left id <$>)
+                            . gStrictTry
+                            $ evalGhcEnv (hscEnvWithImportPaths session) $
+                                do
+                                    env <- getSession
+
+                                    -- Install the module pragmas and options
+                                    df <- liftIO $ setupDynFlagsForGHCiLike env $ ms_hspp_opts ms
+
+                                    -- Desperately looking for an import path
+                                    let impPaths0 = envImportPaths session
+                                        impPaths1 = if null (importPaths df) then Nothing else Just (importPaths df)
+                                        impPaths2 = maybeToList $ moduleImportPath nfp modName
+                                        impPaths = fromMaybe impPaths2 $ impPaths0 <> impPaths1
+                                    dbg "importPaths0" impPaths0
+                                    dbg "importPaths1" impPaths1
+                                    dbg "importPaths2" impPaths2
+                                    dbg "importPaths" impPaths
+                                    -- Restore the cradle import paths
+                                    df <- return df{importPaths = impPaths}
```

Comment:
```suggestion
                                    df <- return df{importPaths = fromMaybe (importPaths df) (envImportPaths session)}
```

Reply:
OK, there were some cases where envImportPaths and importPaths did not return anything, so I added moduleImportPath but I have not been able to reproduce the problem so I simplified this section down to you one--liner